src/utils.js

```diff
@@ -1,5 +1,67 @@
-const fetchWeather = cityName => {
-  const API_KEY = "<INSERT_API_KEY>";
-  // TODO: fetch weather forecast from endpoint
-  // from https://openweathermap.org/api
+import axios from 'axios';
+
+export const fetchWeather = (apiType) => (params) => {
+  return new Promise((resolve, reject) => {
```

Comment:
As `axios` is already promise-based, is there a way that you can refactor this so that it is not wrapped in another `Promise`?

Reply:
Yea sure I can update it.
I did that to return early from the data validation so there didn't need to be an unnecessary API call. I'll just call `Promise.reject(errorMessage)` independently on that line.